### ISSUE TITLE
Documentation for error handling

### ISSUE THREAD

**Issue by frexvahi** [DenisCarriere/geocoder#241]
I tried searching for "error", "fail" and "status" using https://geocoder.readthedocs.io/search.html but did not find any useful results.

I want to know what happens when the location lookup fails, either because the network is down, authentication fails, a rate limit is exceeded, or because the address could not be found in the database.

Playing around shows that the geocoder result object has attributes `ok`, `confidence`, `quality` and so on, what do these mean? Do these have a defined meaning across all providers or are they provider-specific?

In short, how do I check for failed lookups and report them to the user?

**Comment by DenisCarriere** [comment]
Agreed, the documentation is a bit outdated and should clarify the error handling.

As for `confidence` score, it's not defined in the documentation, however this value is shown whenever a `bbox` is provided by the provider, it's a simple lookup table with (higher value == `bbox` is smaller == more precisie && lower value == `bbox` is larger == less precise).

Here's the source code that defines it: https://github.com/DenisCarriere/geocoder/blob/master/geocoder/base.py#L240-L259

If you wanted to report errors to the user, I would use the `status_code`/`status` or `ok` fields from the JSON/GeoJSON object.

```
>>> g.json
{'status': u'ZERO_RESULTS', 'ok': False, 'encoding': 'utf-8', 'status_code': 200, 'location': 'feasfsafasefa', 'provider': 'google'}
```

**Comment by frexvahi** [comment]
OK, thanks for that. I see that you catch `SSLError` in `_connect` but not any of the other exceptions that the `requests` library can raise.
